libs/utils/serialization.py

```diff
@@ -0,0 +1,151 @@
+from collections import namedtuple, Iterable, OrderedDict
+import numpy as np
+import json
+
+from energy_model import (ActiveState, EnergyModelNode, EnergyModelRoot,
+                          PowerDomain, EnergyModel)
+
+def serialize(data):
```

Comment:
I'm not sure there is any value in supporting that manually compared to using yaml.dump. The output will be as readable and zero boilerplate to maintain.